qiskit_nature/problems/second_quantization/lattice/models/fermi_hubbard_model.py

```diff
@@ -0,0 +1,148 @@
+            onsite_interaction: The strength of the on-site interaction.
+        Returns:
+            The Fermi-Hubbard model with uniform parameters.
+        """
+        graph = lattice.graph
+        for node_a, node_b, _ in graph.weighted_edge_list():
+            if node_a == node_b:
+                pass
+            else:
+                graph.update_edge(node_a, node_b, uniform_hopping)
+
+        for node_a in graph.node_indexes():
+            if graph.has_edge(node_a, node_a):
+                graph.update_edge(node_a, node_a, uniform_onsite_potential)
+            else:
+                graph.add_edge(node_a, node_a, uniform_onsite_potential)
+
+        return cls(Lattice(graph), onsite_interaction)
+
+    def second_q_ops(self, display_format: Optional[str] = None) -> FermionicOp:
+        """Return the Hamiltonian of the Fermi-Hubbard model in terms of `FermionicOp`.
+
+        Args:
+            display_format: If sparse, the label is represented sparsely during output.
+                If dense, the label is represented densely during output. Defaults to "dense".
+
+        Returns:
+            FermionicOp: The Hamiltonian of the Fermi-Hubbard model.
+        """
+        kinetic_ham = []
+        interaction_ham = []
+        weighted_edge_list = self._lattice.weighted_edge_list
+        register_length = 2 * self._lattice.num_nodes
+        # kinetic terms
+        for spin in range(2):
+            for node_a, node_b, weight in weighted_edge_list:
+                if node_a == node_b:
+                    index = 2 * node_a + spin
+                    kinetic_ham.append((f"N_{index}", weight))
+
+                else:
+                    if node_a < node_b:
+                        index_left = 2 * node_a + spin
+                        index_right = 2 * node_b + spin
+                        kinetic_ham.append((f"+_{index_left} -_{index_right}", weight))
+                        kinetic_ham.append(
+                            (f"-_{index_left} +_{index_right}", -np.conjugate(weight))
+                        )
+                    elif node_a > node_b:
+                        index_left = 2 * node_b + spin
+                        index_right = 2 * node_a + spin
+                        kinetic_ham.append(
+                            (f"+_{index_left} -_{index_right}", np.conjugate(weight))
+                        )
+                        kinetic_ham.append((f"-_{index_left} +_{index_right}", -weight))
+
+        # on-site interaction terms
+        for node in self._lattice.nodes:
+            index_up = 2 * node
+            index_down = 2 * node + 1
+            interaction_ham.append((f"N_{index_up} N_{index_down}", self.onsite_interaction))
+
+        ham = kinetic_ham + interaction_ham
+
+        return FermionicOp(ham, register_length=register_length, display_format=display_format)
+
+    @classmethod
+    def from_parameters(
+        cls, hopping_matrix: np.ndarray, onsite_interaction: float
```

Comment:
```suggestion
        cls, hopping_matrix: np.ndarray, onsite_interaction: complex
```
Or is there a reason against this case?

Reply:
Out of personal taste I would group the `uniform_parameters` and `from_parameters` classmethods next to each other in the file and move `second_q_ops` either before or after them.